packages/roosterjs-editor-plugins/lib/plugins/TableSelection/TableCellSelection.ts

```diff
@@ -0,0 +1,850 @@
+        this.editor = null;
+    }
+
+    /**
+     * Handle events triggered from editor
+     * @param event PluginEvent object
+     */
+    onPluginEvent(event: PluginEvent) {
+        if (this.editor) {
+            switch (event.eventType) {
+                case PluginEventType.ExtractContentWithDom:
+                    clearSelectedTableCells(event.clonedRoot);
+                    break;
+                case PluginEventType.BeforeCutCopy:
+                    this.handleBeforeCutCopy(event);
+                    break;
+                case PluginEventType.MouseUp:
+                    this.handleMouseUpEvent(event);
+                    break;
+                case PluginEventType.MouseDown:
+                    if (!this.startedSelection) {
+                        this.handleMouseDownEvent(event);
+                    }
+                    break;
+                case PluginEventType.KeyDown:
+                    if (!this.startedSelection) {
+                        this.handleKeyDownEvent(event);
+                    } else {
+                        event.rawEvent.preventDefault();
+                    }
+                    break;
+                case PluginEventType.KeyUp:
+                    if (!this.startedSelection) {
+                        this.handleKeyUpEvent(event);
+                    } else {
+                        event.rawEvent.preventDefault();
+                    }
+                    break;
+                case PluginEventType.Scroll:
+                    if (this.startedSelection) {
+                        event.rawEvent.preventDefault();
+                    }
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Handles the Before Copy Event.
+     * Clear the selection range from the cloned Root.
+     * @param event plugin event
+     */
+    private handleBeforeCutCopy(event: BeforeCutCopyEvent) {
+        const clonedTable = event.clonedRoot.querySelector('table.' + TABLE_SELECTED);
+        if (clonedTable) {
+            const clonedVTable = new VTable(clonedTable as HTMLTableElement);
+
+            clonedVTable.removeCellsBySelection();
+            clonedVTable.writeBack();
+
+            event.range.selectNode(clonedTable);
+
+            if (event.isCut) {
+                this.vTable.forEachSelectedCell(cell => {
+                    if (cell?.td) {
+                        const deleteRange = new Range();
+                        deleteRange.selectNodeContents(cell.td);
+                        deleteRange.deleteContents();
+                        cell.td.appendChild(this.editor.getDocument().createElement('BR'));
+                    }
+                });
+            }
+        }
+        clearSelectedTableCells(event.clonedRoot);
+    }
+
+    //#region Key events
+    /**
+     * Handles the on key event.
+     * @param event the plugin event
+     */
+    private handleKeyDownEvent(event: PluginKeyDownEvent | PluginKeyUpEvent) {
```

Comment:
It seems some of the code from this function can be replaced with Region API.

const regions = editor.getSelectedRegions();
Then check if there are multiple regions, and if they are in the same table.